### PR TITLE
245 check storage area at node pump is sufficient empty pump cellar in some time not one timestep

### PR DESCRIPTION
**Pull request by elisallenens**
This check raises a WARNING if a pumpstation is configured such that it will empty its storage in less than one simulation timestep, since that can lead to simulation instabilities.

### CODE REVIEW

File: threedi_modelchecker/checks/other.py [hunk 1]
```diff
@@ -504,3 +504,38 @@ def get_position(point, linestring):
...
+
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        return (
+            session.query(models.Pumpstation)
+            .join(
+                models.ConnectionNode,
+                models.Pumpstation.connection_node_start_id == models.ConnectionNode.id,
+            )
+            .filter(
+                (models.ConnectionNode.storage_area != None)
```
Comment (caspervdw) [inline threedi_modelchecker/checks/other.py:520]: Not necessary, any comparison to None will result in False anyway

File: threedi_modelchecker/checks/other.py [hunk 2]
```diff
@@ -504,3 +504,38 @@ def get_position(point, linestring):
...
+                        (
+                            models.ConnectionNode.storage_area
+                            * (
+                                models.Pumpstation.start_level
+                                - models.Pumpstation.lower_stop_level
+                            )
+                        )
+                        / models.Pumpstation.capacity
+                    )
+                    < Query(models.GlobalSetting.sim_time_step)
```
Comment (caspervdw) [inline threedi_modelchecker/checks/other.py:532]: This should compare against a scalar subquery of the *first* global setting, not the global setting with id=1. See how it is done with `kmax` in config.py.

File: threedi_modelchecker/checks/other.py [hunk 3]
```diff
@@ -504,3 +504,38 @@ def get_position(point, linestring):
...
+                & (
+                    (  # calculate how many seconds the pumpstation takes to empty its storage: (storage * height)/pump capacity
+                        (
+                            models.ConnectionNode.storage_area
+                            * (
+                                models.Pumpstation.start_level
+                                - models.Pumpstation.lower_stop_level
+                            )
+                        )
+                        / models.Pumpstation.capacity
```
Comment (caspervdw) [inline threedi_modelchecker/checks/other.py:530]: Probably better to switch the capacity to the other side of the `<` to avoid zero division.

So you compare `storage_area * (start_level) - lower_stop_level` against `sim_time_step * capacity`.

File: threedi_modelchecker/tests/test_checks_other.py [hunk 4]
```diff
@@ -366,3 +367,26 @@ def test_potential_breach_interdistance_other_channel(session):
     )
     invalid = check.get_invalid(session)
     assert len(invalid) == 0
+
+
+@pytest.mark.parametrize(
+    "storage_area,time_step,expected_result",
```
Comment (caspervdw) [inline threedi_modelchecker/tests/test_checks_other.py:373]: Can you also add a test for `storage_area = None` and pump capacity `capacity = 0`?

**Review by caspervdw — CHANGES_REQUESTED** [review]

File: threedi_modelchecker/tests/test_checks_other.py [hunk 5]
```diff
@@ -366,3 +367,28 @@ def test_potential_breach_interdistance_other_channel(session):
...
     invalid = check.get_invalid(session)
     assert len(invalid) == 0
+
+
+@pytest.mark.parametrize(
+    "storage_area,time_step,expected_result,capacity",
+    [
+        (0.64, 30, 1, 12.5),
+        (600, 30, 0, 12.5),
+        (None, 30, 1, 12.5),
```
Comment (caspervdw) [inline threedi_modelchecker/tests/test_checks_other.py:377]:
```suggestion
        (None, 30, 0, 12.5),  # no storage --> open water --> no check
```

**Review by caspervdw — APPROVED** [review]
